Framework/Algorithms/src/AddAbsorptionWeightedPathLengths.cpp

```diff
@@ -0,0 +1,162 @@
+
+namespace {
+
+constexpr int DEFAULT_NEVENTS = 1000;
+constexpr int DEFAULT_SEED = 123456789;
+} // namespace
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void AddAbsorptionWeightedPathLengths::init() {
+  declareProperty(
+      std::make_unique<WorkspaceProperty<PeaksWorkspace_sptr::element_type>>(
+          "InputWorkspace", "", Direction::InOut),
+      "An input/output peaks workspace that the path distances will be added "
+      "to.");
+  auto positiveInt = std::make_shared<Kernel::BoundedValidator<int>>();
+  positiveInt->setLower(1);
+  declareProperty("EventsPerPoint", DEFAULT_NEVENTS, positiveInt,
+                  "The number of \"neutron\" events to generate per peak");
+  declareProperty("SeedValue", DEFAULT_SEED, positiveInt,
+                  "Seed the random number generator with this value");
+  declareProperty("MaxScatterPtAttempts", 5000, positiveInt,
+                  "Maximum number of tries made to generate a scattering point "
+                  "within the sample. Objects with "
+                  "holes in them, e.g. a thin annulus can cause problems "
+                  "if this number is too low.\n"
+                  "If a scattering point cannot be generated by increasing "
+                  "this value then there is most likely a problem with "
+                  "the sample geometry.");
+}
+
+std::map<std::string, std::string>
+AddAbsorptionWeightedPathLengths::validateInputs() {
+  PeaksWorkspace_sptr inputWS = getProperty("InputWorkspace");
+  std::map<std::string, std::string> issues;
+  Geometry::IComponent_const_sptr sample =
+      inputWS->getInstrument()->getSample();
+  if (!sample) {
+    issues["InputWorkspace"] = "Input workspace does not have a Sample";
+  } else {
+    if (inputWS->sample().hasEnvironment()) {
+      issues["InputWorkspace"] = "Sample must not have a sample environment";
+    }
+
+    if (inputWS->sample().getMaterial().numberDensity() == 0) {
+      issues["InputWorkspace"] =
+          "Sample must have a material set up with a non-zero number density";
+    }
+  }
+  return issues;
+}
+
+//----------------------------------------------------------------------------------------------
+/** Execute the algorithm.
+ */
+void AddAbsorptionWeightedPathLengths::exec() {
+
+  const PeaksWorkspace_sptr inputWS = getProperty("InputWorkspace");
+  const int nevents = getProperty("EventsPerPoint");
+  const int maxScatterPtAttempts = getProperty("MaxScatterPtAttempts");
+
+  auto instrument = inputWS->getInstrument();
+  auto beamProfile = createBeamProfile(*instrument, inputWS->sample());
+
+  const auto npeaks = inputWS->getNumberPeaks();
+
+  // Configure progress
+  Progress prog(this, 0.0, 1.0, npeaks);
+  prog.setNotifyStep(0.01);
+  const std::string reportMsg = "Computing path lengths";
+
+  // Configure strategy
+  const int nlambda = 1;
```

Comment:
Could this be `constexpr`?

Reply:
Yes I think it could\should. I'll change this and move it to the top of the unit next to a couple of other constexpr declarations that are already there